### DISCUSSION TITLE
OpenSatKit tutorial

### DISCUSSION THREAD

**Discussion by nguyentuanngoc21** [OpenSatKit/OpenSatKit#89] [Q&A] [answered]
If I am totally new with satellite control software in general and OpenSatKit in specific, which tutorial should I follow up to?
Thanks anyway.

**Comment by dmccomas** [comment, marked as answer]
Welcome to the world of satellites!  It can be a bit overwhelming and as
you'll see by this reply I have some work to do in OSK to streamline the
information. In the OSK quick start guide I have a slide that outlines a
roadmap to the documentations (see below). The quickstart guide can be
accessed from the main screen's 'Open Resource"  as shown below.  There are
tutorials that can be launched from some of the screens but unfortunately
the docs and tutorials are not yet organized into nicely flowing lessons.

OSK is focused on flight software.  If you're interested in learning about
satellites here's a website to get you started:  https://spacesteps.com/



[image: image.png]

[image: image.png]

**Reply by nguyentuanngoc21** [reply]
thank you, sir

**Reply by nguyentuanngoc21** [reply]
anyway, I was reading the guidance documents of OSK and saw this 
![image](https://user-images.githubusercontent.com/82352476/167994225-39e0c97a-a8b3-4d69-b5d8-c4bdba047d63.png)
but when I watched the tutorial to create Hello app in OSK, I saw no step to include similar code. I guessed it is Step 2 - edit Cmake, is that right? Since, from what I read, that step is used to include the app I want to use, only that.

**Reply by dmccomas** [reply]
The document you have open is cFE App Developer's Guide.  The first step in the "Manage Create App Workflow" has a screen that let's you select between different styles of "Hello World" app. The "cFE App Dev Guide" selection shown below in screen #1 will generate code that is similar to what you've pointed out in the dev guide. After you select "Create App"  screen #2 below shows the popup window that should appear and it will provide the location of where the code was generated. I named my app 'example' and I'm showing a portion of the OpenSatKit-master/cfs/apps/example//fsw/src/example_app.h that looks like the App Dev Guide example.

**Screen #1**
![ScreenHunter 638](https://user-images.githubusercontent.com/1598728/168826156-e0b7793a-6e43-4588-b791-166e5e0783fc.png)

**Screen #2** 
![ScreenHunter 639](https://user-images.githubusercontent.com/1598728/168826297-af65b3ff-3bbc-4504-8c7f-2cf9f6760441.png)

**Screen #3** 
![ScreenHunter 640](https://user-images.githubusercontent.com/1598728/168826369-222d1096-7737-4b26-be34-117fac14360e.png)

**Reply by nguyentuanngoc21** [reply]
oh,thank you very much
